src/updatedownloader.cpp

```diff
@@ -177,6 +179,14 @@ void UpdateDownloader::Run()
       UpdateDownloadSink sink(*this, tmpdir);
       DownloadFile(m_appcast.DownloadURL, &sink, this);
       sink.Close();
+
+      if (!SignatureVerifier::DSASHA1SignatureValid(sink.GetFilePath(), m_appcast.DsaSignature))
```

Comment:
This block should only be called if a public key is set. This is the promised comment on how I think setting the public key should be done better:

1. Only store it in `Settings`, don't do anything extra.
2. Here, obtain the key from `Settings:: GetDSAPubKeyPem()`. If not empty (but regardless of whether `m_appcast.DsaSignature` is empty or not, because that one is untrusted), call this block, otherwise don't. 
3. Pass the public key as argument to `DSASHA1SignatureValid`, or set it here, rather than doing that at some earlier point.

That would have the advantage that it would also work without OpenSSL DLLs if signatures are not used (which as far as I can tell is not the case with the current code due to setting the key possibly earlier?).

Reply:
OK, I got the idea (basically we keep backward compatibility)
> don't do anything extra

Maybe I'll remove key caching, but replace it with key verification
It will also address `side effect` issue